### ISSUE TITLE
Cannot login - known good password is returning error "▶ ERROR Login required: login failed after passphrase verified"

### ISSUE THREAD

**Issue by kurtroberts** [keybase/client#16523]
I just updated macOS to 10.14.3.  When I opened keybase, it showed me as logged out on this machine, so I tried logging in. It failed (both GUI and CLI) with the following error:
▶ ERROR Login required: login failed after passphrase verified

It's a weirdly phrased message, but just to be sure I opened up my linux machine, and ran `keybase passphrase recover`.  I reset my passphrase, and returned to the Mac. Attempting to log in again gives me exactly the same error.

Keybase GUI Version: 3.1.0-20190307221744+1ebfca5cab

**Comment by maxtaco** [comment]
`keybase log send`

**Comment by maxtaco** [comment]
This is likely a macOS keychain bug ---  does it work to restart your computer btw?

**Comment by kurtroberts** [comment]
Thanks - 400963a37dfcac91eb46281c for the log id, restarting now.

**Comment by kurtroberts** [comment]
After another reboot, I get the same error, on a differently formatted login screen in the app:

There was an error provisioning
login failed after passphrase verified Error code 201: LOGIN_REQUIRED in method keybase.1.login.login

**Comment by maxtaco** [comment]
I think you hit an old bug from 2017 and we can't unlock your local private key.  I think you might need to revoke this device and reprovision it from one of your other devices. Sorry about this!

**Comment by maxtaco** [comment]
Actually, scratch the above, that's not the case.

Did you do any sort of cloud recovery when you brought this machine back up?  It seems like the local keyring doesn't have a key we're expecting.

**Comment by kurtroberts** [comment]
I didn't - I specifically have keychain iCloud sync disabled in system preferences.  I only have icloud configured to sync keynote.

I'm happy to reprovision this device, of course, we can't try that for 5 days because I did the passphrase recovery thing this morning and I'm in a probationary period now, right?

**Comment by maxtaco** [comment]
I cleared that for you, it should work right now.  I was also thinking, would you be willing to run a little node script locally so I can see which keys are in your local keyring?  That might be useful to try a few more debugging ideas.

**Comment by kurtroberts** [comment]
Sure, I mean I’ll read it first and make a final decision, but yeah, happy to help debug!

**Comment by maxtaco** [comment]
Did you wind up reprovisioning the device btw?

**Comment by kurtroberts** [comment]
Yes - I had some work in git I had to finish yesterday afternoon... I hoped that wouldn't destroy too much troubleshooting evidence.

**Comment by maxtaco** [comment]
Maybe, what I was wondering is which keys are stored in your personal keyring. This script will output just the public keys:

```js
const purepack = require('purepack');
const fs = require('fs');
fs.readFile(process.argv[2], 'utf8', (err, data) => {
  var b = Buffer.from(data, 'base64')
  var unpacked = purepack.unpack(b);
  for (i in unpacked) {
     console.log(unpacked[i].body.pub.toString('hex'));
  }
})
```
If you put it in a file `x.js`, then run `node x.js /Users/hyxf/Library/Application Support/Keybase/secretkeys.kurtroberts.mpack`.  Public keys start with the prefixes `0120` and `0121` (and `0101` for PGP).

**Comment by kurtroberts** [comment]
Looks like there are 4 keys that start with `0120` or `0121`:
```
$ node x.js "$HOME/Library/Application Support/Keybase/secretkeys.kurtroberts.mpack" | cut -b 1-5
0120d
01218
c6c14
c6c14
01201
0121e
```

If you need more of the output, can I keybase it to you?

**Comment by maxtaco** [comment]
Sure thing, thanks!  They're public on your keybase profile, but it would be useful to see them all.  I'm max at keybase, thanks Kurt!

Everything's work OK now btw after the reprovision? Sorry about this bug BTW

**Comment by kurtroberts** [comment]
Yes, everything's working great again!  I use the service on a couple devices each day, so I appreciate the quick help too!

**Comment by maxtaco** [comment]
Great, glad to hear it.  We'll dig into that bug, got the debug data on our side.
